ConcurrentAPI.adoc

```diff
@@ -0,0 +1,244 @@
+== Distributed Concurrent API
+
+=== Motivation
+
+Coming from an initial issue where we need to provide an implementation for vert.x cluster manager for the shared data feature,
+provide concurrent and distributed programming capabilities.
+
+* Distributed Locks
+* Distributed CountDownLatch
+
+=== Scope
+
+All these interfaces should be placed in commons, in order to be used either in Embedded and Server mode.
+
+== Infinispan Distributed Locks
+
+=== DistributedLockManager and configuration
+
+```java
+package org.infinispan.lock.api;
+
+public class DistributedLockManager {
+   boolean defineLock(LockConfiguration configuration);
+   InfinispanLock get(String name);
```

Comment:
well, for me this should behave like counters somethow. Why not making everything non blocking. I don't think this is a big issue concerning vert.x